streaming/src/main/scala/org/apache/spark/streaming/util/FileBasedWriteAheadLog.scala

```diff
@@ -205,10 +205,12 @@ private[streaming] class FileBasedWriteAheadLog(
 
   /** Stop the manager, close any open log writer */
   def close(): Unit = synchronized {
-    if (currentLogWriter != null) {
-      currentLogWriter.close()
+    if (!executionContext.isShutdown) {
```

Comment:
Ah! do you actually mean checking if `shutdown` is idempotent. Let me check out.